File: Weno/Ju/Convection.jl
module Convection
export flux,minf,maxf
function flux(x::Float64)
    x
end
function minf(a::Float64,b::Float64)
    if b<=0.0
        return flux(b)
    elseif a>=0.0
        return flux(a)
    else
        return 0.0
    end
end
function maxf(a::Float64,b::Float64)
    max(flux(a),flux(b))
end
end

File: Weno/PyVec/Convection.py
import numpy as np

def flux(x):
    return x

def minf(a,b):
#    if b<=0:
#      return flux(b)
#    elif a>=0:
#      return flux(a)
#    else:
#      return 0.0
    return (b <= 0) * flux(b) + (a >= 0) * flux(a)

def maxf(a,b):
    return np.maximum(flux(a), flux(b))
